src/Corporation/Division.ts

```diff
@@ -35,6 +35,9 @@ export class Division {
 
   products = new JSONMap<string, Product>();
   makesProducts = false;
+  get maxProducts() {
```

Comment:
What is the purpose of this getter? IMO, `getMaximumNumberProducts` is a fine name, there's no reason to have two names for the same thing.